woocommerce/woocommerce-gutenberg-products-block
# Store Customization > Create a new endpoint within Store API for updating patterns with AI-generated content

## What

Create a new endpoint within Store API for updating patterns with AI-managed images fetched from the Verticals API.

### `POST wp-json/wc/store/patterns`

- The endpoint is accessible exclusively to users with `manage_options` capabilities.
- The request is allowed only if the store owner granted permissions during the onboarding phase (only if `'woocommerce_blocks_allow_ai_connection'` is `true` ).
- The endpoint has an optional `'business_description'` parameter, which corresponds to a string with a brief description for the business: if not provided, the API will automatically fetch whatever is saved within the `'woo_ai_describe_store_description'` wp_option to generate the content that will be used to populate the patterns.

🎯 #10658

## Why



This is a requirement to ensure all of our patterns can have AI-managed images and, later on, AI-generated content.

## Testing Instructions



_Please consider any edge cases this change may have, and also other areas of the product this may impact._

1. Create a new [JN install](https://jurassic.ninja/) with both Jetpack and Woocommerce installed and activated.
3. Install and activate WooCommerce Blocks from this zip file:  [woocommerce-gutenberg-products-block.zip](https://github.com/woocommerce/woocommerce-blocks/files/12598602/woocommerce-gutenberg-products-block.zip)

5. In your wp-admin, you should see the following section for Jetpack:

<img width="1803" alt="Screenshot 2023-09-11 at 18 55 38" src="https://github.com/woocommerce/woocommerce-blocks/assets/15730971/fc804759-f724-4fa2-829d-7bfd441962ec">

6. Click on "Set up Jetpack" and make sure the connection is successful:

<img width="1994" alt="Screenshot 2023-09-11 at 18 58 30" src="https://github.com/woocommerce/woocommerce-blocks/assets/15730971/661ee4c5-a0ac-40fd-9e5e-835a0e4c1cb7">


7. Using [Postman](https://www.postman.com/) or any other equivalent app, make a POST request to the following endpoint: `wp-json/wc/store/patterns` (e.g. `https://outside-dwelling.jurassic.ninja/wp-json/wc/store/patterns`). Since you are not authenticated, you should get the following error:

<img width="748" alt="Screenshot 2023-09-13 at 16 25 45" src="https://github.com/woocommerce/woocommerce-blocks/assets/15730971/fee21120-392a-4c7f-abe7-e70a3999ff32">

8. Now, we need to make sure the request works for admin users: you can do so by adding your credentials via Postman directly or, alternatively:

- From your admin screen, click on Tools > Plugin editor: 

<img width="337" alt="Screenshot 2023-09-11 at 18 34 23" src="https://github.com/woocommerce/woocommerce-blocks/assets/15730971/c353bd13-bf00-4bc8-93d6-d827ca99997f">

- Select the WooCommerce Blocks plugin and add the following within your woocommerce-blocks/woocommerce-gutenberg-products-block.php file:

```php
function connect_to_endpoint() {
	update_option( 'woocommerce_blocks_allow_ai_connection', true );

	$request = new WP_REST_Request( 'POST', '/wc/store/patterns' );

	$request->set_param( 'business_description', 'selling art' );

	$response = rest_do_request( $request )->get_data();

	error_log( print_r( $response, true ) );
}

add_action( 'admin_init', 'connect_to_endpoint' );
```
- Now access your JN install via SSH (you can copy the command for access over here:)

<img width="346" alt="Screenshot 2023-09-11 at 18 38 44" src="https://github.com/woocommerce/woocommerce-blocks/assets/15730971/419f338f-d203-4bec-8f78-4589d5ad3ad2">

- Head over to your wp-content directory and tail the debug.log file. Example:

```
cd /srv/users/user964902c9/apps/{insert-your-jurassic-ninja-username-here}/public/wp-content
tail -f debug.log
```

- Now go back to your browser and refresh the page in the main admin page e.g. `https://your-site.jurassic.ninja/wp-admin`

- Go back to your terminal and confirm that you can see the following response:

<img width="475" alt="Screenshot 2023-09-13 at 16 43 52" src="https://github.com/woocommerce/woocommerce-blocks/assets/15730971/95917fdf-c55d-457a-a4ed-e3f08a1179d3">

* [] Do not include in the Testing Notes 

## WooCommerce Visibility


Required:


## Checklist

Required:

Conditional:

## Changelog


N/A
